Jasmine/LevelDesignerViewController.swift

```diff
@@ -0,0 +1,108 @@
+import UIKit
+
+class LevelDesignerViewController: UIViewController {
+
+    @IBOutlet private var levelNameField: UITextField!
+
+    /// The order of the game mode on the gameModeControl
+    private static let gameModeOrder: [GameMode] = [.tetris, .sliding, .swapping]
+    @IBOutlet private var gameModeControl: UISegmentedControl!
+
+    /// The order of the game type on the gameTypeControl
+    private static let gameTypeOrder: [GameType] = [.chengYu, .ciHui]
+    @IBOutlet private var gameTypeControl: UISegmentedControl!
+
+    /// The formatted text for the select phrases button, with %d representing the phrases count
+    private static let selectPhrasesButtonText = "Select phrases (%d currently)"
+    @IBOutlet private var selectPhrasesButton: UIButton!
+
+    /// The levels data to add to or update. To be set by LevelSelectionViewController
+    var levels: Levels?
+    /// The level under edit. To be set by LevelSelectionViewController
+    var levelToEdit: Level?
+    /// The phrases that has been selected. To be set to PhrasesExplorerViewModel's phrases
+    private var selectedPhrases: Phrases? {
+        didSet {
+            let buttonText = String(format: LevelDesignerViewController.selectPhrasesButtonText,
+                                    selectedPhrases?.count ?? 0)
+            selectPhrasesButton.setTitle(buttonText, for: .normal)
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setDataFromLevelToEdit()
+    }
+
+    /// Set the controls and texts to display based on the level to edit
+    private func setDataFromLevelToEdit() {
+        guard let level = levelToEdit else {
+            return
+        }
+        levelNameField.text = level.name
+
+        guard let gameModeIndex = LevelDesignerViewController.gameModeOrder.index(of: level.gameMode),
+              let gameTypeIndex = LevelDesignerViewController.gameTypeOrder.index(of: level.gameType) else {
+            fatalError("Unable to get game mode/type index")
+        }
+        gameModeControl.selectedSegmentIndex = gameModeIndex
+        gameTypeControl.selectedSegmentIndex = gameTypeIndex
+
+        selectedPhrases = level.phrases
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let phrasesExplorerViewController = segue.destination as? PhrasesExplorerViewController {
+            // TODO : initialize VM and set selected phrases (waiting for Phrases and PhrasesExplorerViewModel update)
+        }
+    }
+
+    // TODO : wait for level select VC to be implemented and segue to it
+    @IBAction private func onSaveAndCloseButtonPressed(_ button: UIButton) {
+        saveGame()
```

Comment:
I think can just simply dismiss this view. Or are you intending to tell the previous vc some information?